File: src/main/java/com/cctrader/indicators/technical/MovingAverageExponentialConvergence.java
package com.cctrader.indicators.technical;

import com.cctrader.data.MarketDataSet;

/**
 * returns a positive value if fast is above slow.
 * <p/>
 * Normalize with price normalizer?? or max min in last period
 */
public class MovingAverageExponentialConvergence {
    final int slow;
    final int fast;

    public MovingAverageExponentialConvergence(int slow, int fast) {
        this.slow = slow;
        this.fast = fast;
    }

    public double get(int index, MarketDataSet marketDataSet) {
        double slowEMAVC = ExponentialMovingAverage.get(index, slow, marketDataSet);
        double fastEMAVC = ExponentialMovingAverage.get(index, fast, marketDataSet);

        return fastEMAVC - slowEMAVC;
    }
}

File: src/main/scala/com/cctrader/indicators/technical/MovingAverageExponentialConvergence.scala
package com.cctrader.indicators.technical

import com.cctrader.data.MarketDataSet
import com.cctrader.indicators.InputIndicator

/**
 *
 */
class MovingAverageExponentialConvergence (fastN: Int, slowN: Int) extends InputIndicator{
  /**
   * Calculating the indicator.
   *
   * @param t index of time now in the marketDataSet
   * @param data the marketDataSet
   * @return
   */
  override def apply(t: Int, data: MarketDataSet): Double = {
    val fastMovingAverage = new ExponentialMovingAveragePrice(fastN)
    val slowMovingAverage = new ExponentialMovingAveragePrice(slowN)

    (fastMovingAverage.apply(t, data) - slowMovingAverage.apply(t, data))
  }

}